src/PKSim.Presentation/Presenters/Main/MenuAndToolBarPresenter.cs

```diff
@@ -516,32 +514,33 @@ public void Visit(SensitivityAnalysis sensitivityAnalysis)
 
       public void Visit(ParameterIdentification parameterIdentification)
       {
-         _currentlyActivePI = parameterIdentification.Id;
-         _parameterIdentificationRunning = _activePIs.Contains(parameterIdentification.Id);
          updateParameterIdentificationItems(parameterIdentification);
          _eventPublisher.PublishEvent(new ParameterIdentificationSelectedEvent(parameterIdentification));
       }
 
       public void Handle(ParameterIdentificationStartedEvent parameterIdentificationEvent)
       {
-         _activePIs.Add(parameterIdentificationEvent.ParameterIdentification.Id);
-         _parameterIdentificationRunning = true;
-         updateParameterIdentificationItems(parameterIdentificationEvent.ParameterIdentification);
+         var parameterIdentification = parameterIdentificationEvent.ParameterIdentification; 
+         _runningParameterIdentifications.Add(parameterIdentification);
+         updateParameterIdentificationItems(parameterIdentification);
       }
 
       public void Handle(ParameterIdentificationTerminatedEvent parameterIdentificationEvent)
       {
-         _activePIs.Remove(parameterIdentificationEvent.ParameterIdentification.Id);
-         if (parameterIdentificationEvent.ParameterIdentification.Id == _currentlyActivePI)
-            _parameterIdentificationRunning = false;
-         updateParameterIdentificationItems(parameterIdentificationEvent.ParameterIdentification);
+         var parameterIdentification = parameterIdentificationEvent.ParameterIdentification;
+         _runningParameterIdentifications.Remove(parameterIdentification);
+         //only update if selected subject in the actual parameter identification terminated
+         var activeSubject = _activeSubjectRetriever.Active<ParameterIdentification>();
+         if(Equals(activeSubject, parameterIdentification))
```

Comment:
you were always updating the view. I think this was wrong, It should only be done when the active subject is the one also terminated no?